pkg/instrumentation/podmutator.go

```diff
@@ -104,14 +111,25 @@ func (pm *instPodMutator) Mutate(ctx context.Context, ns corev1.Namespace, pod c
 	}
 	insts.DotNet = inst
 
+	if inst, err = pm.getInstrumentationInstance(ctx, ns, pod, annotationInjectGolang); err != nil {
+		// we still allow the pod to be created, but we log a message to the operator's logs
+		logger.Error(err, "failed to select an OpenTelemetry Instrumentation instance for this pod")
+		return pod, err
+	}
+	if EnableGolangAutoInstrumentationSupport.IsEnabled() {
+		insts.Golang = inst
+	} else {
+		logger.Error(err, "support for Golang auto instrumentation is not enable")
```

Comment:
I'm trying to understand what the resulting behavior would be here if someone attempts to use golang auto inst if the flag hasn't been enabled yet. Would they receive a mutated pod with some of the added defaults, but not have the added sidecar from the mutator? Should we return an error here instead so someone would be informed they need to enable the feature gate before they can use this feature?

Reply:
My intention was that if a pod adds the `annotationInjectGolang` annotation and the operator has not enabled the feature gate then their pod is not mutated at all for Goland.  I didn't return an error bc I wanted other  `annotationInject*` annotations on the pod to still be respected.